core/src/main/scala/kafka/coordinator/transaction/TransactionStateManager.scala

```diff
@@ -278,36 +281,40 @@ class TransactionStateManager(brokerId: Int,
     val topicPartition = new TopicPartition(Topic.TRANSACTION_STATE_TOPIC_NAME, partitionId)
 
     inLock(stateLock) {
+      leavingPartitions.remove(partitionId)
```

Comment:
Re: check `leavingPartitions`, sounds good, will add it.

Re: lock protection during loading, as discussed offline, I will make `loadingPartitions` volatile.

Re: coordinatorEpoch, good point, ack.